content/webapp/services/prismic/transformers/index.ts

```diff
@@ -130,6 +157,324 @@ type PromoImage = {
   superWidescreenImage?: ImageType;
 };
 
+function transformTitledTextItem(item) {
+  return {
+    title: parseTitle(item.title),
+    text: parseStructuredText(item.text),
+    link: parseLink(item.link),
+    label: isDocumentLink(item.label) ? parseLabelType(item.label) : null,
+  };
+}
+
+export function transformBody(body: Body): BodyType {
```

Comment:
bikeshed: this is mostly a function which applies appropriate transformations to discriminated unions. In some places it does that transformation itself, in others it calls out to some function that does the work (like [here](https://github.com/wellcomecollection/wellcomecollection.org/pull/7679/files#diff-92d364df89c68fd24a7e8dc0fbccbe376ca3b11bce51619b9a82032f117f8d7aR242-R257). Should we make it consistently like that, so that all of the `switch`/`case`s look the same?

Reply:
I like this suggestion, but for now I'm going to leave it as-is – switching to TypeScript is a big enough change with bug potential; a more invasive refactor is even riskier.

I've left a TODO comment to come back and look at this properly, once we're all-in on TypeScript and can really lean on the type checker to catch our mistakes.